ssh_key_rotator/publisher/key_publisher.py

```diff
@@ -65,21 +65,39 @@ def publish_new_key(self) -> str:
         return public_key.decode()
 
 
+def create_argument_parser() -> ArgumentParser:
+    argument_parser = ArgumentParser(
+        prog="key_publisher",
+        description="Creates public/private SSH keys and publishes the public key either locally or to S3 (default is S3)",
+        epilog="Thanks for using key_publisher! :)",
+    )
+
+    parser.add_argument("user")
+    parser.add_argument("hostname")
+    parser.add_argument("-ds", "--datastore", choices=["s3", "local"], default="s3")
+
+    options = parser.add_argument_group("Options")
+    options.add_argument(
+        "-ds (s3 | local)",
+        "--datastore (s3 | local)",
+        help="choose where to store the public key, on S3 or on the local system (default is S3)"
+    )
+
+    arguments = parser.add_argument_group("Required Arguments")
+    arguments.add_argument("user")
+    arguments.add_argument("hostname")
+
+    return argument_parser
+
+
 if __name__ == "__main__":
-    command_line_arguments = sys.argv
-    user_flag_index = command_line_arguments.index("--user")
-    username = command_line_arguments[user_flag_index + 1]
-    server_flag_index = command_line_arguments.index("--server")
-    server = command_line_arguments[server_flag_index + 1]
-    data_store_flag_index = command_line_arguments.index("--datastore")
-    data_store = command_line_arguments[data_store_flag_index + 1]
-
-    if data_store == "local":
-        publisher = LocalPublisher(server, username)
+    parser = create_argument_parser()
+    args = parser.parse_args()
+
+    if args.datastore == "local":  # If the user chose to store the public key locally
+        publisher = LocalPublisher(args.hostname, args.user)
         publisher.publish_new_key()
-    elif data_store == "s3":
-        s3_bucket_name = input("Enter a bucket name: ")
-        publisher = S3Publisher(s3_bucket_name, server, username)
+    else:  # If the user chose to store the public key on S3 or chose to default to S3
+        s3_bucket_name = input("Enter an S3 bucket name: ")
```

Comment:
This should be a command line arg incorporated into argparser. otherwise looks good Im not too familair with the module itself